includes/core/server/request/class-update-intention.php

```diff
@@ -0,0 +1,196 @@
+<?php
+/**
+ * Class file for WCPay\Core\Server\Request\Update_Intent.
+ *
+ * @package WooCommerce Payments
+ */
+
+namespace WCPay\Core\Server\Request;
+
+use WC_Payments;
+use WCPay\Core\Exceptions\Server\Request\Invalid_Request_Parameter_Exception;
+use WCPay\Core\Server\Request;
+use WC_Payments_API_Client;
+use WCPay\Payment_Methods\CC_Payment_Method;
+use WCPay\Payment_Methods\Link_Payment_Method;
+
+/**
+ * Request class for creating intents.
+ */
+class Update_Intention extends Request {
+	use Intention;
+	use Level3;
+
+	const IMMUTABLE_PARAMS = [ 'amount' ];
+	const REQUIRED_PARAMS  = [ 'amount', 'currency' ];
+	const DEFAULT_PARAMS   = [
+		'receipt_email' => '',
+		'metadata'      => [],
+	];
+
+	/**
+	 * Intent id.
+	 *
+	 * @var string|null $intent_id
+	 */
+	private $intent_id = null;
+
+	/**
+	 * Set intent id.
+	 *
+	 * @param string $intent_id Intent id.
+	 *
+	 * @return void
+	 * @throws Invalid_Request_Parameter_Exception
+	 */
+	public function set_intent_id( string $intent_id ) {
+		$this->validate_stripe_id( $intent_id, [ 'pi' ] );
+
+		// Prevent mutation of intent id. It can be only set once.
+		if ( null === $this->intent_id ) {
+			$this->intent_id = $intent_id;
+		}
+	}
+
+	/**
+	 * Returns the request's API.
+	 *
+	 * @return string
+	 * @throws Invalid_Request_Parameter_Exception
+	 */
+	public function get_api(): string {
+		if ( null === $this->intent_id ) {
+			throw new Invalid_Request_Parameter_Exception( __( 'Intent ID is not set.', 'woocommerce-payments' ), 'wcpay_core_request_intent_not_set' );
+		}
+		return WC_Payments_API_Client::INTENTIONS_API . '/' . $this->intent_id;
```

Comment:
At this point we're not returning an API here, but a URL instad. Should we rename the method altogether?

Reply:
Yeah, we could. Maybe we can rename it to `get_url` or `get_endpoint`, `get_uri`, or `get_path` (or something similar) ?

Reply:
Ah, let's just leave it like this for now. I'm looking at the API client's `request` method, and it already uses `$api` there, maybe it's not worth changing.